simularium/simularium-website
# Feature/record movie UI

Time estimate or Size
=======
_Review time: medium_

Problem
=======
Closes #327 

Solution
========
`RecordMovieComponent` manages the modal and button, retrieves and formats the various display icons and text, defines the downloading function, and calls the viewer handlers it receives.

Modal is trim, no logic.

Minimized storing state in `ViewerPanel`: only holding the url for the blob.


* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
On supported browsers:
1. Load a trajectory, click record, confirm that recording indicator shows up next to button.
2. During playback the record button should pulse red unless hovering, when it should show a red square.
3. Click stop recording, and then confirm download via the modal.
4. Check that the downloaded movie is playable.

On unsupported browsers  (Firefox, and apparently sometimes Safari):
1. Confirm that the button has disabled styling and does nothing when clicked.

Screenshots (optional):
-----------------------
![Screenshot 2024-03-26 at 10 41 52 AM](https://github.com/simularium/simularium-website/assets/24981838/14254085-228e-4471-b140-4eac792baf8a)
![Screenshot 2024-03-26 at 10 41 46 AM](https://github.com/simularium/simularium-website/assets/24981838/96c12b9c-ab9a-4c05-a963-4491282bd61d)
![Screenshot 2024-03-26 at 10 41 40 AM](https://github.com/simularium/simularium-website/assets/24981838/74b18c5d-cde4-401b-883d-f7d716237802)
![Screenshot 2024-03-26 at 10 41 34 AM](https://github.com/simularium/simularium-website/assets/24981838/7bfe288f-8ff2-4b82-b476-6aacb97b128b)
